extensions/libarchive/tests/util/ArchiveTests.cpp

```diff
@@ -86,46 +87,43 @@ void build_test_archive(const std::filesystem::path& path, const TAE_MAP_T& entr
 
     std::cout << "Adding entry: " << name << std::endl;
 
-    archive_entry_set_filetype(entry, test_entry.type);
-    archive_entry_set_pathname(entry, test_entry.name.c_str());
-    archive_entry_set_size(entry, gsl::narrow<la_int64_t>(test_entry.size));
-    archive_entry_set_perm(entry, test_entry.perms);
-    archive_entry_set_uid(entry, test_entry.uid);
-    archive_entry_set_gid(entry, test_entry.gid);
-    archive_entry_set_mtime(entry, test_entry.mtime, test_entry.mtime_nsec);
+    archive_entry_set_filetype(entry.get(), test_entry.type);
+    archive_entry_set_pathname(entry.get(), test_entry.name.c_str());
+    archive_entry_set_size(entry.get(), gsl::narrow<la_int64_t>(test_entry.size));
+    archive_entry_set_perm(entry.get(), test_entry.perms);
+    archive_entry_set_uid(entry.get(), test_entry.uid);
+    archive_entry_set_gid(entry.get(), test_entry.gid);
+    archive_entry_set_mtime(entry.get(), test_entry.mtime, test_entry.mtime_nsec);
 
-    archive_write_header(test_archive, entry);
-    archive_write_data(test_archive, test_entry.content, test_entry.size);
+    archive_write_header(test_archive.get(), entry.get());
+    archive_write_data(test_archive.get(), test_entry.content, test_entry.size);
 
-    archive_entry_clear(entry);
+    archive_entry_clear(entry.get());
   }
-
-  archive_entry_free(entry);
-  archive_write_close(test_archive);
 }
 
-void build_test_archive(const std::filesystem::path& path, OrderedTestArchive& ordered_archive) {
+void build_test_archive(const std::filesystem::path& path, const OrderedTestArchive& ordered_archive) {
   build_test_archive(path, ordered_archive.map, ordered_archive.order);
 }
 
 bool check_archive_contents(const std::filesystem::path& path, const TAE_MAP_T& entries, bool check_attributes, const FN_VEC_T& order) {
   FN_VEC_T read_names;
   FN_VEC_T extra_names;
   bool ok = true;
-  struct archive *a = archive_read_new();
+  auto a = minifi::processors::archive_read_unique_ptr{archive_read_new()};
   struct archive_entry *entry = nullptr;
 
-  archive_read_support_format_all(a);
-  archive_read_support_filter_all(a);
+  archive_read_support_format_all(a.get());
+  archive_read_support_filter_all(a.get());
 
-  int r = archive_read_open_filename(a, path.string().c_str(), 16384);
+  int r = archive_read_open_filename(a.get(), path.string().c_str(), 16384);
```

Comment:
And this

Reply:
👍 [review changes](https://github.com/apache/nifi-minifi-cpp/pull/1880/commits/3e748c09d8bd829d5e2b968ba9fda7108ae8f311)